From 9bfee6442a35eae60eceb821086970ad59a4a9cc Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Fri, 22 Jun 2012 07:02:25 +0200
Subject: [PATCH] xilinx_wdt: Hack SETTIMEOUT

Just to keep busybox watchdog happy. Timeout must be more than 30 seconds.

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/watchdog/of_xilinx_wdt.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/watchdog/of_xilinx_wdt.c b/drivers/watchdog/of_xilinx_wdt.c
index a10b0bc..c8b9b6a 100644
--- a/drivers/watchdog/of_xilinx_wdt.c
+++ b/drivers/watchdog/of_xilinx_wdt.c
@@ -262,6 +262,9 @@ static long xwdt_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 		xwdt_keepalive();
 		return 0;
 
+	case WDIOC_SETTIMEOUT:
+		printk(KERN_INFO "This feature is not implemented yet!\n");
+
 	case WDIOC_GETTIMEOUT:
 		if (no_timeout)
 			return -ENOTTY;
-- 
1.7.0.4

